x509/x509.go

```diff
@@ -352,6 +371,7 @@ var signatureAlgorithmDetails = []struct {
 	{ECDSAWithSHA256, oidSignatureECDSAWithSHA256, ECDSA, crypto.SHA256},
 	{ECDSAWithSHA384, oidSignatureECDSAWithSHA384, ECDSA, crypto.SHA384},
 	{ECDSAWithSHA512, oidSignatureECDSAWithSHA512, ECDSA, crypto.SHA512},
+	{ED25519SIG, oidKeyEd25519, ED25519, crypto.Hash(0)},
```

Comment:
It would be nice to have something like `crypto.NoDigest` instead of manually creating `crypto.Hash(0)`...this is consistent with the existing `MD2WithRSA` case, but the reasoning seems to be different: presumably the MD2 case is using crypto.Hash(0) because MD2 just isn't supported, rather than there actually being no hash.

Reply:
I didn't realize the `cryptyo.Hash(0)` in MD2 was do to lack of support, so I definitely agree that clarifying the reasoning would be useful. I can add a `cryptoNoDigest` var above (with some documentation explaining that it should be used if/when there is no hash).